types/react-select/src/types.d.ts

```diff
@@ -64,18 +64,18 @@ export interface CommonProps<
     options: OptionsType<OptionType>;
     selectOption: (option: OptionType) => void;
     selectProps: SelectProps<OptionType, IsMulti, GroupType>;
-    setValue: (newValue: ValueType<OptionType, IsMulti>, action: SetValueAction, option: OptionType) => void;
+    setValue: (newValue: ValueType<OptionType, IsMulti>, action: SetValueAction, option?: OptionType) => void;
 }
 
 export interface SelectOptionActionMeta<OptionType extends OptionTypeBase> {
     action: 'select-option';
-    option: OptionType;
+    option: OptionType | undefined;
```

Comment:
Out of curiosity: Why `undefined` vs. `key?:`

Reply:
I usually use `undefined` when the key is specified but the value can be `undefined` and I use `?` when the key may or may not be defined. In this case the `option` key is always specified (see [code](https://github.com/JedWatson/react-select/blob/master/packages/react-select/src/Select.js#L797)), but its value might be `undefined`.